app/uk/gov/hmrc/customs/declaration/model/fileTransmissionModels.scala

```diff
@@ -66,3 +66,63 @@ object FileTransmission {
   implicit val urlFormat = HttpUrlFormat
   implicit val writes = Json.writes[FileTransmission]
 }
+
+sealed trait FileTransmissionOutcome {
+  val outcome: String
+}
+case object FileTransmissionSuccessOutcome extends FileTransmissionOutcome {
+  override val outcome: String = "SUCCESS"
+}
+case object FileTransmissionFailureOutcome extends FileTransmissionOutcome {
+  override val outcome: String = "FAILURE"
+}
+
+object FileTransmissionOutcome {
+  implicit val readsFileTransmissionOutcome: Reads[FileTransmissionOutcome] = new Reads[FileTransmissionOutcome] {
+    override def reads(json: JsValue): JsResult[FileTransmissionOutcome] = json match {
+      case JsString(FileTransmissionSuccessOutcome.outcome) => JsSuccess(FileTransmissionSuccessOutcome)
+      case JsString(FileTransmissionFailureOutcome.outcome) => JsSuccess(FileTransmissionFailureOutcome)
+      case _ => JsError(s"Invalid FileTransmissionOutcome $json")
+    }
+  }
+}
+
+sealed trait FileTransmissionNotification {
+  val fileReference: FileReference
+  val batchId: BatchId
+  val outcome: FileTransmissionOutcome
+}
+
+case class FileTransmissionSuccessNotification(fileReference: FileReference,
+                                               batchId: BatchId,
+                                               outcome: FileTransmissionOutcome = FileTransmissionSuccessOutcome
+                                    ) extends FileTransmissionNotification
+
+object FileTransmissionSuccessNotification {
+  implicit val readsSuccessCallback: Reads[FileTransmissionSuccessNotification] = Json.reads[FileTransmissionSuccessNotification]
+}
+
+case class FileTransmissionFailureNotification(fileReference: FileReference,
+                                               batchId: BatchId,
+                                               outcome: FileTransmissionOutcome = FileTransmissionFailureOutcome,
```

Comment:
now that you have a decider and 2 concrete type - do you still need optional `errorDetails: Option[String]`? I think this can be just `errorDetails: String`

Reply:
Done